aten/src/ATen/native/cuda/LogAddExpKernel.cu

```diff
@@ -13,22 +13,45 @@
 namespace at::native {
 
 void logaddexp_kernel_cuda(TensorIteratorBase& iter) {
-  AT_DISPATCH_FLOATING_TYPES_AND2(
-      ScalarType::BFloat16, ScalarType::Half,
-      iter.dtype(), "logaddexp_cuda",
-      [&]() {
-        using opmath_t = at::opmath_type<scalar_t>;
-        gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a_, scalar_t b_) -> scalar_t {
-          const auto a = static_cast<opmath_t>(a_);
-          const auto b = static_cast<opmath_t>(b_);
-          if (::isinf(a) && a == b) {
-            return a;
-          } else {
-            const auto m = ::max(a, b);
-            return m + ::log1p(::exp(-::abs(a - b)));
-          }
+  auto dtype = iter.dtype();
+  if(isComplexType(dtype)) {
```

Comment:
You should be able to merge these outer level branches. `real()` will not produce an error when the underlying dtype is real, so the logic in this first branch should be correct for all dtypes.